### ISSUE TITLE
Temporal Fields: 'NoneType' object has no attribute 'upper'

### ISSUE THREAD

**Issue by alanmcruickshank** [apache/incubator-superset#2556]
### Superset version
0.17.1

### Expected results
Temporal fields with `None` as their `Type` should either function the same as ` ` as their `Type` or not be created at all.

### Actual results
Slice fails to load with error `'NoneType' object has no attribute 'upper'`

Trace:

```
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/superset/viz.py", line 265, in get_payload
    df = self.get_df()
  File "/usr/local/lib/python2.7/site-packages/superset/viz.py", line 109, in get_df
    self.results = self.datasource.query(query_obj)
  File "/usr/local/lib/python2.7/site-packages/superset/connectors/sqla/models.py", line 525, in query
    sql = self.get_query_str(engine, qry_start_dttm, **query_obj)
  File "/usr/local/lib/python2.7/site-packages/superset/connectors/sqla/models.py", line 429, in get_query_str
    time_filter = dttm_col.get_time_filter(from_dttm, to_dttm)
  File "/usr/local/lib/python2.7/site-packages/superset/connectors/sqla/models.py", line 68, in get_time_filter
    col >= text(self.dttm_sql_literal(start_dttm)),
  File "/usr/local/lib/python2.7/site-packages/superset/connectors/sqla/models.py", line 117, in dttm_sql_literal
    self.type, dttm)
  File "/usr/local/lib/python2.7/site-packages/superset/db_engine_specs.py", line 239, in convert_dttm
    if target_type.upper() in ('DATETIME', 'DATE'):
AttributeError: 'NoneType' object has no attribute `'upper'
```

### Steps to reproduce
Create an additional field on a table and mark it as temporal. The `Type` will default to `None` rather than ` ` as per previous versions. This cases a slice using this as the `Time Column` to fail with the above error.

Two possible ways to fix: Either default back to ` ` (blank)  rather than `None` or `NULL`, or add in some handling for `None` values in `convert_dttm`.

**Comment by mistercrunch** [comment]
Fixed in 0.17.3

**Comment by mistercrunch** [comment]
Notice: this issue has been closed because it has been inactive for 382 days. Feel free to comment and request for this issue to be reopened.
